ballerina/utils.bal

```diff
@@ -71,3 +71,87 @@ isolated function getChatMessageStringContent(Prompt|string prompt) returns stri
     }
     return str.trim();
 }
+
+isolated function entryMatchesFilters(VectorMatch|InMemoryVectorEntry entry, MetadataFilters filters) returns boolean|error {
+    Metadata? metadata = entry.chunk.metadata;
+    if metadata is () {
+        return false;
+    }
+    return check evaluateFilterNode(metadata, filters);
+}
+
+isolated function evaluateFilterNode(Metadata content, MetadataFilters|MetadataFilter node) returns boolean|error {
+    if node is MetadataFilter {
+        return content.hasKey(node.key) ? compareValues(content.get(node.key), node.operator, node.value) : false;
+    }
+    boolean[] results = [];
+    foreach MetadataFilters|MetadataFilter child in node.filters {
+        boolean childResult = check evaluateFilterNode(content, child);
+        results.push(childResult);
+    }
+    return evaluateCondition(node.condition, results);
+}
+
+isolated function evaluateCondition(MetadataFilterCondition condition, boolean[] results) returns boolean {
+    if condition == AND {
+        foreach boolean result in results {
+            if !result {
+                return false;
+            }
+        }
+        return true;
+    }
+    foreach boolean result in results {
```

Comment:
You can use `some` operation

Reply:
FIxed by [259f1de](https://github.com/ballerina-platform/module-ballerina-ai/pull/70/commits/259f1de42fb36cbae55e4f0eb8fd01357d37aed4)